lib/health.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const request = require('request-promise-native');
+
+const hasMatchingMethod = (option, reqMethod) => {
+  return option.methods.some(method => {
+    return method.toLowerCase() === reqMethod.toLowerCase();
+  });
+};
+
+module.exports = endpoints => {
+
+  return (req, res, next) => {
+
+    if (endpoints && endpoints.length) {
+
+      let requests = endpoints.reduce((memo, option) => {
+        // Test the method property against the current req method
+        if (option.methods && (hasMatchingMethod(option, req.method) !== true)) {
+          return memo;
+        }
+        // memoize the endpoints
+        memo.push(request(option.url).promise());
+        return memo;
+      }, []);
```

Comment:
1) This initial PR defaults to make the health requests on all methods (i.e., when the form loads, and when the form is submitted). The `methods` option gives the user the flexibility to decide which methods the request should be invoked on. However, I am open to the suggestion that it should default to only `GET` requests.

2) I was considering a `steps` option: The intention being that a user could optionally supply the middleware with a list of steps that they want the health request to be made on. 
E.g.
```js
health: [{
   url: 'api.domain.com/monitor',
   steps: ['start', 'confirm'],
}]
```

I would like opinions